### PR TITLE
Export bboxes dataset as VIA-tracks file

### PR DESCRIPTION
**Pull request by sfmig**
Before submitting a pull request (PR), please read the [contributing guide](https://github.com/neuroinformatics-unit/.github/blob/main/CONTRIBUTING.md).

Please fill out as much of this template as you can, but if you have any problems or questions, just leave a comment and we will help out :)

## Description

**What is this PR**

- [ ] Bug fix
- [ ] Addition of a new feature
- [ ] Other

**Why is this PR needed?**

**What does this PR do?**

## References

Please reference any existing issues/PRs that relate to this PR.

## How has this PR been tested?

Please explain how any new code has been tested, and how you have ensured that no existing functionality has changed.

## Is this a breaking change?

If this PR breaks any existing functionality, please explain how and why.

## Does this PR require an update to the documentation?

If any features have changed, or have been added. Please explain how the
documentation has been updated.

## Checklist:

- [ ] The code has been tested locally
- [ ] Tests have been added to cover all new functionality
- [ ] The documentation has been updated to reflect any changes
- [ ] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

**Review by niksirbi — APPROVED** [review]
Looking good @sfmig!

I'm approving this for your convenience.
I left some comments, many of them optional.

My main two quibbles are:
- I find the name and description of the `extract_track_id_from_individuals` arguments confusing (I couldn't understand what exactly it was doing before reading the source code). I've left some suggestions for alternative ways to describe it.
- Some error messages are not logged

**Comment by niksirbi** [comment]
Note: this PR will need to be rebased and updated after #606 is merged, mostly because the Input/Output guide has been restructured.

File: movement/io/save_bboxes.py [hunk 1]
```diff
@@ -0,0 +1,543 @@
...
+            "The requested number of digits to represent the frame "
+            "number cannot be used to represent all the frame numbers."
+            f"Got {frame_n_digits}, but the maximum frame number has "
+            f"{min_required_digits} digits"
+        )
+    else:
+        return frame_n_digits
+
+
+def _get_map_individuals_to_track_ids(
```
[the remark this answers is not in the document]
Comment (sfmig) [reply movement/io/save_bboxes.py]: I used map to refer to the dictionary that the function returns. I renamed it to "compute_...map" for clarity

File: tests/test_unit/test_io/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
+"""Unit tests for the movement.io.utils module."""
```
[the remark this answers is not in the document]
Comment (sfmig) [reply tests/test_unit/test_io/test_utils.py]: I moved the tests here under `test_validators/test_files_validator` following your previous suggestion on removing the `utils.py` module (otherwise the test modules won't mirror the "functionality" modules).

As far as I can tell, the added tests are testing features that were not tested before (valid file, file that already exists, file with invalid suffix, file path with multiple valid suffixes). The one with more overlap with existing code would be the file permissions test. However in the added test we do it without mocking, which is arguably more realistic. 

Note that we did have a "file with invalid suffix" test before, but it is just checking a specific case of invalid suffix (input with '.txt', expected suffix one of '.csv', '.h5'). The added test is more comprehensive (it also tests wrong inputs to the suffixes argument, such as None or empty strings).

My aim here was to add tests for some aspects I saw were missing, but probably there is an underlying need to revisit the validation tests? It does feel like that would deserve its own PR. I don't think it is priority for now.

Let me know if you'd like me to remove the added tests entirely, if you think that makes things clearer.

File: tests/test_unit/test_io/test_save_bboxes.py [hunk 3]
```diff
@@ -0,0 +1,751 @@
...
+    if image_file_prefix is not None:
+        assert df["filename"].str.startswith(image_file_prefix).all()
+    else:
+        assert df["filename"].str.startswith("0").all()
+
+    # Check image file suffix is as expected
+    if image_file_suffix is not None:
+        assert df["filename"].str.endswith(image_file_suffix).all()
+    else:
+        assert df["filename"].str.endswith(".png").all()
```
[the remark this answers is not in the document]
Comment (sfmig) [reply tests/test_unit/test_io/test_save_bboxes.py:193]: Line 193 checks that the filename entries in the dataframe end with the default `.png` when the parametrised variable `image_file_suffix = None`. `.all()` is the dataframe method that checks all the values in the dataframe (rather than all the parametrisation values).

File: tests/test_unit/test_io/test_utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (niksirbi) [reply tests/test_unit/test_io/test_utils.py]: I agree with the decisions you took here, and I'm grateful for the additional tests.
This part of the tests could probably use some refactoring in the future, but I agree it's absolutely not a priority any time soon.

**Comment by niksirbi** [comment]
Thanks for implementing the suggestions @sfmig, I think this PR is now good to go, except for 1 final minor point:

The `test_nwb.py` file can also be moved into the `test_io` folder. The tested `nwb.py` file is under `io` and it also belongs there conceptually.
